idea/frontend/src/features/sails/hooks/use-sails.ts

```diff
@@ -0,0 +1,42 @@
+import { HexString } from '@gear-js/api';
+import { useAlert } from '@gear-js/react-hooks';
+import { useQuery } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { Sails } from 'sails-js';
+
+import { RPCError, RPCErrorCode } from '@/shared/services/rpcService';
+
+import { getIdl } from '../api';
+
+function useSails(codeId: HexString | null | undefined) {
+  const alert = useAlert();
+
+  const getSails = async () => {
+    if (!codeId) throw new Error('Code ID is not found');
+
+    const { result } = await getIdl(codeId);
+    const sails = (await Sails.new()).parseIdl(result);
```

Comment:
no need to initialise new Sails instance every time. It can be initialised once during app loading.